dependencydiff/mapping.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2022 Security Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dependencydiff
+
+import (
+	"fmt"
+)
+
+// EcosystemOSV is a package ecosystem supported by OSV.
+type ecosystemOSV string
+
+// Data source: https://ossf.github.io/osv-schema/#affectedpackage-field
+const (
+	// The Go ecosystem.
+	goOSV ecosystemOSV = "Go"
+
+	// The NPM ecosystem.
+	npmOSV = "npm"
+
+	// The Android ecosystem
+	androidOSV = "Android" // nolint:unused
+
+	// The crates.io ecosystem for RUST.
+	cratesOSV = "crates.io"
+
+	// For reports from the OSS-Fuzz project that have no more appropriate ecosystem.
+	ossFuzzOSV = "OSS-Fuzz" // nolint:unused
+
+	// The Python PyPI ecosystem. PyPI is the main package source of pip.
+	pyPIOSV = "PyPI"
+
+	// The RubyGems ecosystem.
+	rubyGemsOSV = "RubyGems"
+
+	// The PHP package manager ecosystem. Packagist is the main Composer repository.
+	packagistOSV = "Packagist"
+
+	// The Maven Java package ecosystem.
+	mavenOSV = "Maven"
+
+	// The NuGet package ecosystem.
+	nuGetOSV = "Nuget"
+
+	// The Linux kernel.
+	linuxOSV = "Linux" // nolint:unused
+
+	// The Debian package ecosystem.
+	debianOSV = "Debian" // nolint:unused
+
+	// Hex is the package manager of Erlang.
+	// TODO: GitHub doesn't support hex as the ecosystem for Erlang yet. Add this mapping in the future.
+	hexOSV = "Hex" // nolint:unused
+)
+
+// ecosystemGitHub is a package ecosystem supported by GitHub.
+type ecosystemGitHub string
+
+// Data source: https://docs.github.com/en/code-security/supply-chain-security/understanding-
+// your-software-supply-chain/about-the-dependency-graph#supported-package-ecosystems
+// nolint
+const (
```

Comment:
Can we simply this to just a hash map? We just care about a mapping between our ecosystem and a string from GH, and the hashmap is clear enough for the reader.

Reply:
done & resolved, thanks!